### ISSUE TITLE
Workflow executionOrder v1 is not working as expected in n8n version 1.22.4

### ISSUE THREAD

**Issue by nenetchi** [n8n-io/n8n#8281]
**Describe the bug**
Workflow with executionOrder v1 shows v0 behavior.

**To Reproduce**

[TestExecOrder.json](https://github.com/n8n-io/n8n/files/13884512/TestExecOrder.json)

![image](https://github.com/n8n-io/n8n/assets/9377596/0a0f4bde-15a4-4eb7-a013-21fb2e026cfa)

- Node in false branch executed
- The `date` command in `Execute Top` and `Execute Bottom` print same time

**Expected behavior**
- The topmost branch should be run first.
- The node in `false` branch should not be triggered by the Merge node.

**Environment (please complete the following information):**

- OS: Debian 10.13
- n8n Version: 1.22.4
- Node.js Version: v18.18.2
- Database system: Postgres 14.10-1.pgdg120+1
- Operation mode: queue

**Comment by janober** [comment]
Very strange. I did just test locally and for me it did execute like this:
![Screenshot from 2024-01-10 08-55-15](https://github.com/n8n-io/n8n/assets/6249596/1d3051f1-56df-4ffc-8ef7-ca6537ce7759)

Meaning totally correctly. Only when I set "Execution Order" to "v0" does it display in the same order as you described above. Have also not seen a problem like that before and nobody else ever reported problems with the logic since the v1 release.

Can you please go in the settings again and set to "v0" and then again "v1" just to make sure it got really set correctly to v1.

**Comment by nenetchi** [comment]
Hey @janober, thanks for your quick response. 🙏

I tried setting it to v0 then v1, delete and reimport the workflow, restart the container, but the problem still exists.

**Comment by nenetchi** [comment]
This bug only happens when I trigger the workflow with production webhook URL. When I click the `Execute Workflow` button and trigger the workflow with testing webhook URL, the bug disappeared.

**Comment by janober** [comment]
Thanks a lot. That piece of information is very helpful. We will have a look what is going on there.

**Comment by Joffcom** [comment]
The internal dev ticket for this is `N8N-7157`

**Comment by nenetchi** [comment]
After some experiments, I solved the bug with this patch.

https://github.com/n8n-io/n8n/blob/78416d8b83c0d5a1f2729ea767afd728533ef98a/packages/cli/src/databases/repositories/execution.repository.ts#L216-L227

```
diff --git a/packages/cli/src/databases/repositories/execution.repository.ts b/packages/cli/src/databases/repositories/execution.repository.ts
index 093430326..8170ca947 100644
--- a/packages/cli/src/databases/repositories/execution.repository.ts
+++ b/packages/cli/src/databases/repositories/execution.repository.ts
@@ -217,10 +217,10 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 		const { data, workflowData, ...rest } = execution;
 		const { identifiers: inserted } = await this.insert(rest);
 		const { id: executionId } = inserted[0] as { id: string };
-		const { connections, nodes, name } = workflowData ?? {};
+		const { connections, nodes, name, settings } = workflowData ?? {};
 		await this.executionDataRepository.insert({
 			executionId,
-			workflowData: { connections, nodes, name, id: workflowData?.id },
+			workflowData: { connections, nodes, name, settings, id: workflowData?.id },
 			data: stringify(data),
 		});
 		return String(executionId);
```

**Comment by krynble** [comment]
Thanks for your report and the fix informatioin, i've submitted a [PR](https://github.com/n8n-io/n8n/pull/8288) for this.

**Comment by Joffcom** [comment]
Hey @nenetchi,

The fix for this was released in `1.24.0`, I am going to mark this one as closed and if you still see an issue let me know and we can open this again.
